### PR TITLE
CRDCDH-641

### PR DESCRIPTION
**Pull request by knockknockyoo**

### CODE REVIEW

File: src/metadata_validator.py [hunk 1]
```diff
@@ -231,14 +231,87 @@ def validate_props(self, dataRecord, model):
...
-        return {VALIDATION_RESULT: result, ERRORS: errors, WARNINGS: warnings}
-    
+        result = {"result": STATUS_ERROR, ERRORS: [], WARNINGS: []}
+        if not data_record.get("parents"):
+            result["result"] = STATUS_WARNING
+            result[WARNINGS].append(create_error(FAILED_VALIDATE_RECORDS, "Parent property does not exist or empty"))
+            return result
+
+        data_record_parent_nodes = data_record.get("parents")
+        nodes = node_definition[MODEL].get("nodes", {})
```
Comment (vshand11) [inline src/metadata_validator.py:262]: the node definition passed in is an instance of Model class.  To access mode dict, use node_definition.model

**Review by vshand11 — CHANGES_REQUESTED** [review]
Hi Young, finished the first round of review.   please take a look on comments.

File: src/metadata_validator.py (hunk 1 above)
Comment (knockknockyoo) [reply src/metadata_validator.py]: removed and integrated with the model class

File: src/common/mongo_dao.py [hunk 2]
```diff
@@ -43,6 +43,31 @@ def get_submission(self, submissionId):
...
+
+    """
+    check node exists by node name and its value
+    """
+    # def validate_node(self, db_name, submission_id, node_type, node_key, node_value):
+    def search_nodes_by_type_and_value(self, nodes):
+        db = self.client[self.db_name]
+        data_collection = db[DATA_COLlECTION]
+        try:
+            node_set, query = set(), []
```
[the remark this answers is not in the document]
Comment (knockknockyoo) [reply src/common/mongo_dao.py]: relocated the precursor.

**Comment by vshand11** [comment]
Hi Young,  find one issue at metadata_validator when run debugging mode.  At line 278, node_definition_relationships = model.get_node_relationships(parent_type),   the param, "parent_type" should be dataRecord["nodeType"].  Please take a look.

**Comment by knockknockyoo** [comment]
> Hi Young, find one issue at metadata_validator when run debugging mode. At line 278, node_definition_relationships = model.get_node_relationships(parent_type), the param, "parent_type" should be dataRecord["nodeType"]. Please take a look.

Yes, I've adjusted as you requested.

**Review by vshand11 — APPROVED** [review]
The new version passed review and passed local testing

**Pull request merged**